src/pudl/analysis/record_linkage/eia_ferc1_record_linkage.py

```diff
@@ -68,39 +345,44 @@ def out_pudl__yearly_assn_eia_ferc1_plant_parts(
             reported aggregated to the FERC1 plant-level.
         out_eia__yearly_plant_parts: The EIA plant parts list.
     """
-    inputs = InputManager(
+    inputs = get_compiled_input_manager(
         out_ferc1__yearly_all_plants,
         out_ferc1__yearly_steam_plants_fuel_by_plant_sched402,
         out_eia__yearly_plant_parts,
     )
-    # compile/cache inputs upfront. Hopefully we can catch any errors in inputs early.
-    inputs.execute()
-    features_all = Features(feature_type="all", inputs=inputs).get_features(
-        clobber=False
-    )
-    features_train = Features(feature_type="training", inputs=inputs).get_features(
-        clobber=False
-    )
-    match_df = run_model(
+    # all_pairs_df = get_all_pairs_df(inputs)
+    # train_pairs_df = get_train_pairs_df(inputs)
+    all_pairs_df, train_pairs_df = get_pairs_dfs(inputs)
+    vectorizer = get_pair_vectorizers()
+    features_all = embed_dataframe.embed_dataframe_graph(all_pairs_df, vectorizer)
+    features_train = embed_dataframe.embed_dataframe_graph(train_pairs_df, vectorizer)
+    y_df = get_y_label_df(train_pairs_df, inputs)
+    match_df = run_matching_model(
         features_train=features_train,
         features_all=features_all,
-        train_df=inputs.train_df,
+        y_df=y_df,
     )
     # choose one EIA match for each FERC record
-    best_match_df = find_best_matches(match_df).pipe(
-        overwrite_bad_predictions, inputs.train_df
-    )
+    best_match_df = get_best_matches_with_overwrites(match_df, inputs)
     # join EIA and FERC columns back on
-    connects_ferc1_eia = prettyify_best_matches(
-        best_match_df,
-        train_df=inputs.get_train_df(),
-        plant_parts_eia_true=inputs.get_plant_parts_eia_true(),
-        plants_ferc1=inputs.get_plants_ferc1(),
-    ).pipe(add_null_overrides)  # Override specified values with NA record_id_eia
-    connects_ferc1_eia = Resource.from_id(
-        "out_pudl__yearly_assn_eia_ferc1_plant_parts"
-    ).enforce_schema(connects_ferc1_eia)
-    return connects_ferc1_eia
+    ferc1_eia_connected_df = get_match_full_records(best_match_df, inputs)
+    return ferc1_eia_connected_df
+
+
+@asset(
+    name="out_pudl__yearly_assn_eia_ferc1_plant_parts",
+    io_manager_key="pudl_sqlite_io_manager",
+    compute_kind="Python",
+)
+def out_pudl__yearly_assn_eia_ferc1_plant_parts(
+    _out_pudl__yearly_assn_eia_ferc1_plant_parts: pd.DataFrame,
+) -> pd.DataFrame:
+    """Linkage between FERC1 plants and EIA plant parts for persistent storage.
+
+    Args:
+        out_eia__yearly_plant_parts: The EIA plant parts list.
+    """
+    return _out_pudl__yearly_assn_eia_ferc1_plant_parts
```

Comment:
@zschira I found out that you can't write an output to persistent storage from a `graph_asset` so I made this separate `asset` which takes the intermediary output from the `graph_asset` `_out_pudl__yearly_assn_eia_ferc1_plant_parts` and basically just outputs as an asset to the DB. Maybe there's a better way to do this besides having this wrapper `asset` which is almost identically named? It's nice to keep the graph asset though so you can run individual ops, and not have to run the whole linking process at once.

Reply:
I think the problem is that we're not setting the io manager in the graph asset. I remember messing with this awhile back for something, and I think I could only get it to work by using a `graph_multi_asset` because the API is a little weird. I can see if I can get it to work that way in the morning, and if not it's probably ok as is

Reply:
Hmm I think I tried setting it in the `graph_multi_asset` and didn't have success. I think `graph_multi_asset` doesn't take `io_manager_key` as a parameter, and so I tried to set it within the `outs` parameter:

`outs={"out_pudl__yearly_assn_eia_ferc1_plant_parts" : AssetOut(io_manager_key="pudl_sqlite_io_manager")` 

But maybe that's not the right way to go about it.